src/Data/Text/Display.hs

```diff
@@ -26,405 +14,18 @@ module Data.Text.Display
-deriving via (DisplayDecimal Word8) instance Display Word8
-
--- | @since 0.0.1.0
-deriving via (DisplayDecimal Word16) instance Display Word16
-
--- | @since 0.0.1.0
-deriving via (DisplayDecimal Word32) instance Display Word32
-
--- | @since 0.0.1.0
-deriving via (DisplayDecimal Word64) instance Display Word64
-
--- | @since 0.0.1.0
-deriving via (ShowInstance IOException) instance Display IOException
-
--- | @since 0.0.1.0
-deriving via (ShowInstance SomeException) instance Display SomeException
-
--- | @since 0.0.1.0
-instance (Display a, Display b) => Display (a, b) where
-  displayBuilder (a, b) = "(" <> displayBuilder a <> "," <> displayBuilder b <> ")"
-
--- | @since 0.0.1.0
-instance (Display a, Display b, Display c) => Display (a, b, c) where
-  displayBuilder (a, b, c) = "(" <> displayBuilder a <> "," <> displayBuilder b <> "," <> displayBuilder c <> ")"
-
--- | @since 0.0.1.0
-instance (Display a, Display b, Display c, Display d) => Display (a, b, c, d) where
-  displayBuilder (a, b, c, d) = "(" <> displayBuilder a <> "," <> displayBuilder b <> "," <> displayBuilder c <> "," <> displayBuilder d <> ")"
-
--- $designChoices
---
--- === A “Lawless Typeclass”
---
--- The 'Display' typeclass does not contain any law. This is a controversial choice for some people,
--- but the truth is that there are not any laws to ask of the consumer that are not already enforced
--- by the type system and the internals of the 'Data.Text.Internal.Text' type.
---
--- === "🚫 You should not try to display functions!"
---
--- Sometimes, when using the library, you may encounter this message:
---
--- > • 🚫 You should not try to display functions!
--- >   💡 Write a 'newtype' wrapper that represents your domain more accurately.
--- >      If you are not consciously trying to use `display` on a function,
--- >      make sure that you are not missing an argument somewhere.
---
--- The 'display' library does not allow the definition and usage of 'Display' on
--- bare function types (@(a -> b)@).
--- Experience and time have shown that due to partial application being baked in the language,
--- many users encounter a partial application-related error message when a simple missing
--- argument to a function is the root cause.
---
--- There may be legitimate uses of a 'Display' instance on a function type.
--- But these usages are extremely dependent on their domain of application.
--- That is why it is best to wrap them in a newtype that can better
--- express and enforce the domain.
---
--- === "🚫 You should not try to display ByteStrings!"
---
--- An arbitrary ByteStrings cannot be safely converted to text without prior knowledge of its encoding.
---
--- As such, in order to avoid dangerously blind conversions, it is recommended to use a specialised
--- function such as 'Data.Text.Encoding.decodeUtf8'' or 'Data.Text.Encoding.decodeUtf8With' if you wish to turn a UTF8-encoded ByteString
--- to Text.
+import Data.Text.Display.Core
+import Data.Text.Display.Generic
```

Comment:
How does this translate in terms of haddocks? Will people see barebones re-exports page?

Reply:
forget what I wrote, you fix this at the top it seems.